src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -190,6 +325,30 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
             }
         }
 
+        private void UpdateWatchedSettings(Dictionary<KeyValueWatcher, KeyValueChange> changedKeyValues)
+        {
+            foreach(KeyValuePair<KeyValueWatcher, KeyValueChange> kvp in changedKeyValues)
+            {
+                KeyValueChange keyValueChange = kvp.Value;
+                KeyValueWatcher changeWatcher = kvp.Key;
+                KeyValueIdentifier kvIdentifier = new KeyValueIdentifier(changeWatcher.Key, changeWatcher.Label);
+
+                if (keyValueChange.ChangeType == KeyValueChangeType.Modified)
+                {
+                    _watchedSettings[kvIdentifier] = keyValueChange.Current;
+                }
+                else if (keyValueChange.ChangeType == KeyValueChangeType.Deleted)
+                {
+                    _watchedSettings.Remove(kvIdentifier);
+                }
+                else if (keyValueChange.ChangeType == KeyValueChangeType.Added)

```

Comment:
It looks like this was added for the line `_applicationSettings[kvIdentifier.Key] = keyValueChange.Current;`. This is already handled in ProcessChanges